std/selector/multiplexer.go

```diff
@@ -0,0 +1,98 @@
+package selector
+
+import (
+	"math/big"
+
+	"github.com/consensys/gnark/backend/hint"
+	"github.com/consensys/gnark/frontend"
+)
+
+func init() {
+	// register hints
+	hint.Register(MuxIndicators)
+	hint.Register(MapIndicators)
+}
+
+// Map is a key value associative array: the output will be values[i] such that keys[i] == sel. If keys does not
+// contain sel, no proofs can be generated. If keys array has more than one key that equals to sel, the output will
+// be undefined, and the output could be a linear combination of the corresponding values with those keys.
+//
+// In case keys and values do not have the same length, this function will panic.
+func Map(api frontend.API, sel frontend.Variable,
```

Comment:
What do you think about renaming `sel` to `queryKey` here?

Reply:
Yes, I think it is better for understandability.